fabric-network/src/impl/query/roundrobinqueryhandler.js

```diff
@@ -13,51 +13,46 @@ const util = require('util');
 const logger = require('fabric-network/lib/logger').getLogger('RoundRobinQueryHandler');
 
 class RoundRobinQueryHandler {
-	constructor(peers, options = {}) {
+	constructor(peers) {
 		logger.debug('constructor: peers=%j', peers.map((peer) => peer.name));
 		this._peers = peers;
 		this._currentPeerIndex = 0;
-		this._options = options;
 	}
 
 	async evaluate(query) {
 		const method = 'evaluate';
 		logger.debug('%s - start', method);
 
-		const startPeerIndex = this._currentPeerIndex;
-		const errorMessages = [];
-		const options = {requestTimeout: 3000};
-		// use the timeout as the requestTimeout or let default
-		if (Number.isInteger(this._options.timeout)) {
-			options.requestTimeout = this._options.timeout * 1000; // in ms;
+		if (this._currentPeerIndex >= this._peers.length) {
+			// start over
+			this._currentPeerIndex = 0;
 		}
 
+		const startPeerIndex = this._currentPeerIndex++;
+		const errorMessages = [];
+
 		for (let i = 0; i < this._peers.length; i++) {
 			const peerIndex = (startPeerIndex + i) % this._peers.length;
-			this._currentPeerIndex = peerIndex;
+
 			const peer = this._peers[peerIndex];
-			logger.debug('%s - query sending to peer %s', method, peer.name);
-			try {
-				const results = await query.send({targets:[peer]}, options);
-				if (results && results.errors && results.errors.length > 0) {
-					logger.error('%s - problem with query to peer %s error:%s', method, peer.name, results.errors[0]);
-					// since only one peer, only one error
-					errorMessages.push(results.errors[0].message);
-				} else if (results && results.responses && results.responses[0].response.status < 400) {
-					logger.debug('%s - peer response status %s', method, results.responses[0].response.status);
-					return results.responses[0].response.payload;
-				} else if (results && results.responses && results.responses[0].response.message) {
-					logger.debug('%s - peer response status: %s message: %s', method,
-						results.responses[0].response.status,
-						results.responses[0].response.message);
-					errorMessages.push(results.responses[0].response.message);
+			logger.debug('%s - sending to peer %s', method, peer.name);
+
+			const results = await query.send([peer]);
+			if (results && results[peer.name]) {
+				const result = results[peer.name];
+				if (result instanceof Error) {
+					errorMessages.push(result.toString());
 				} else {
-					logger.error('%s - no results returned from the query', method);
-					errorMessages.push('No response from query');
+					if (result.status < 400) {
+						logger.debug('%s - peer response status: %s', method, result.status);
+						return result.payload;
+					}
+					logger.debug('%s - peer response status: %s message: %s', method, result.status, result.message);
+					errorMessages.push(result.message);
 				}
-			} catch (error) {
-				logger.error('%s - problem with query to peer %s error:%s', method, peer.name, error);
-				errorMessages.push(error.message);
+			} else {
```

Comment:
The `else` branch should not be necessary

Reply:
removed